langchain4j-pgvector/src/main/java/dev/langchain4j/store/embedding/pgvector/PgVectorEmbeddingStore.java

```diff
@@ -186,51 +239,54 @@ public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedde
     }
 
     /**
-     * Finds the most relevant (closest in space) embeddings to the provided reference embedding.
+     * Searches for the most similar (closest in the embedding space) {@link Embedding}s.
+     * <br>
+     * All search criteria are defined inside the {@link EmbeddingSearchRequest}.
+     * <br>
+     * {@link EmbeddingSearchRequest#filter()} is used to filter by meta dada.
      *
-     * @param referenceEmbedding The embedding used as a reference. Returned embeddings should be relevant (closest) to this one.
-     * @param maxResults         The maximum number of embeddings to be returned.
-     * @param minScore           The minimum relevance score, ranging from 0 to 1 (inclusive).
-     *                           Only embeddings with a score of this value or higher will be returned.
-     * @return A list of embedding matches.
-     * Each embedding match includes a relevance score (derivative of cosine distance),
-     * ranging from 0 (not relevant) to 1 (highly relevant).
+     * @param request A request to search in an {@link EmbeddingStore}. Contains all search criteria.
+     * @return An {@link EmbeddingSearchResult} containing all found {@link Embedding}s.
      */
     @Override
-    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+    public EmbeddingSearchResult<TextSegment> search(EmbeddingSearchRequest request) {
+        Embedding referenceEmbedding = request.queryEmbedding();
+        int maxResults = request.maxResults();
+        double minScore = request.minScore();
+        Filter filter = request.filter();
+
         List<EmbeddingMatch<TextSegment>> result = new ArrayList<>();
-        try (Connection connection = setupConnection()) {
+        try (Connection connection = getConnection()) {
             String referenceVector = Arrays.toString(referenceEmbedding.vector());
+            String whereClause = (filter == null) ? "" : metadataHandler.whereClause(filter);
```

Comment:
```suggestion
            String whereClause = (filter == null) ? "" : "WHERE " + metadataHandler.whereClause(filter);
```
then line below can be removed